src/Utils.cpp

```diff
@@ -0,0 +1,76 @@
+#include <yaml-cpp/yaml.h>
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "../include/Utils.h"
+#include "../include/Common.h"
+
+using clang::tooling::runToolOnCode, clang::tooling::FrontendActionFactory,
+    clang::tooling::CommonOptionsParser, clang::tooling::ClangTool,
+    clang::FrontendAction;
+using std::size_t, std::vector, std::string, std::cerr, std::cout, std::endl,
+    std::ifstream;
+
+std::unique_ptr<FrontendActionFactory>
+newBaseFrontendActionFactory(vector<BaseTransformation> transformations,
+                             string output_path) {
+    /* Method is similar to newFrontendActionFactory but this method
+     * supports passing arguments to the FrontendAction constructor */
+    class SimpleFrontendActionFactory : public FrontendActionFactory {
+     public:
+        SimpleFrontendActionFactory(vector<BaseTransformation> transformations, string output_path):
+        transformations(move(transformations)), output_path(move(output_path)) {}
+        FrontendAction *create() override {
+            return new BaseFrontendAction(transformations, output_path);
+        }
+     private:
+        vector<BaseTransformation> transformations;
+        string output_path;
+    };
+
+    return std::unique_ptr<FrontendActionFactory>(
+            new SimpleFrontendActionFactory(move(transformations), move(output_path)));
+}
+
+Runner::Runner(int argc, const char **argv) : argc(argc), argv(argv) {}
+
+vector<BaseTransformation>
+Runner::getTransformations(const string &config_path) {
+  YAML::Node config = YAML::LoadFile(config_path);
+  vector<BaseTransformation> transformations;
+  for (auto transform_data : config["transformations"]) {
+    if (transform_data["basic transform"]) {
```

Comment:
It's better to define vector with all known transformations. And add a private property with a getter for name. After that, you can iterate over known transformation and compare their name to this.